### PR TITLE
api: deprecate tls_context in favor of transport socket

### LINKED ISSUE #8098: v3 API: move tls_context to transport_socket only
Currently we setup raw_buffer / tls transport_socket based on existence of tls_context (partly as transition).

The motivation to move tls_context to transport socket (extension config) is to better support the proposed transport_socket_matcher: https://github.com/envoyproxy/envoy/issues/8016

### PR DESCRIPTION
**Pull request by lizan**
Signed-off-by: Lizan Zhou <lizan@tetrate.io>

Description:
Deprecates `tls_context` in favor of transport socket.

Risk Level: Low
Testing: CI
Docs Changes: Added
Release Notes: Added
Fixes #8098.

### CODE REVIEW

File: configs/envoy_double_proxy_v2.template.yaml [hunk 1]
```diff
@@ -8,17 +8,20 @@
...
-        - certificate_chain:
-            filename: certs/servercert.pem
-          private_key:
-            filename: certs/serverkey.pem
-        validation_context: {}
-        alpn_protocols:
-        - h2
-        - http/1.1
+    transport_socket:
+      name: tls
```
Comment (htuch) [inline configs/envoy_double_proxy_v2.template.yaml]: Should we create an alias `envoy.tls` and use this everywhere?

File: test/extensions/clusters/dynamic_forward_proxy/cluster_test.cc [hunk 2]
```diff
@@ -173,7 +173,7 @@ TEST_F(ClusterTest, InvalidLbContext) {
 }
 
 // Verify that using 'sni' causes a failure.
-TEST_F(ClusterTest, InvalidSNI) {
+TEST_F(ClusterTest, DEPRECATED_FEATURE_TEST(InvalidSNI)) {
```
Comment (htuch) [inline test/extensions/clusters/dynamic_forward_proxy/cluster_test.cc]: What is going to happen to these tests in `v3`? Can we update them now?

Comment (lizan) [reply test/extensions/clusters/dynamic_forward_proxy/cluster_test.cc]: These tests are best effort anyway. (i.e. if a TLS transport socket is set inside TAP transport socket the cluster config won't fail) And the override takes priority anyway even SNI is set in UpstreamTlsContext.

File: configs/envoy_double_proxy_v2.template.yaml (hunk 1 above)
Comment (lizan) [reply configs/envoy_double_proxy_v2.template.yaml]: Yeah I thought about that, do double registration is possible but to do it properly (with deprecation) will result another large PR. We might want to do that sometime anyway but let's keep this for now? `tls` transport socket is there for quite a while.

Comment (htuch) [reply configs/envoy_double_proxy_v2.template.yaml]: Fair enough, can you file an issue?

File: test/extensions/clusters/dynamic_forward_proxy/cluster_test.cc (hunk 2 above)
Comment (htuch) [reply test/extensions/clusters/dynamic_forward_proxy/cluster_test.cc]: I'm still a bit confused here. I think either we are OK with stripping these tests completely at the next major version (with resulting potential coverage loss), or we should fix them up while doing the deprecation. It's not like they are doing something that relies inherently on the deprecated TLS context, I think.

File: configs/envoy_double_proxy_v2.template.yaml (hunk 1 above)
Comment (lizan) [reply configs/envoy_double_proxy_v2.template.yaml]: https://github.com/envoyproxy/envoy/issues/8540

File: test/extensions/clusters/dynamic_forward_proxy/cluster_test.cc (hunk 2 above)
Comment (lizan) [reply test/extensions/clusters/dynamic_forward_proxy/cluster_test.cc]: IMO it's OK to strip these tests at the next major version and the config validation check as well.

**Comment by htuch** [comment]
@lizan needs master merge, good to go otherwise.

File: api/envoy/api/v3alpha/BUILD [hunk 3]
```diff
@@ -7,7 +7,7 @@ licenses(["notice"])  # Apache 2
 api_proto_package(
     has_services = True,
     deps = [
-        "//envoy/api/v3alpha/auth:pkg",
+        "//envoy/api/v2/auth:pkg",
```
Comment (lizan) [inline api/envoy/api/v3alpha/BUILD]: @htuch I merged master and then try to use fix_format to address merge conflicts in v3alpha, then this sneaked in and passes format check. seems like a bug.

Comment (htuch) [reply api/envoy/api/v3alpha/BUILD]: Yeah, this seems a bit off. I'll take a look.

**Comment by htuch** [comment]
@lizan can you merge master and get CI passing? This is ready to go. Thanks.
/wait

**Comment by lizan** [comment]
@htuch this is good to go.

**Comment by htuch** [comment]
@lizan more master merge needed :)

**Comment by lizan** [comment]
@htuch merged

**Pull request merged**